apps/lib/apps.c

```diff
@@ -3196,6 +3196,33 @@ BIO *bio_open_default_quiet(const char *filename, char mode, int format)
     return bio_open_default_(filename, mode, format, 1);
 }
 
+int mem_bio_to_file(BIO *in, const char *filename, int format, int private)
+{
+    int rv = 0, ret = 0;
+    BIO *out = NULL;
+    BUF_MEM *mem_buffer = NULL;
+
+    rv = BIO_get_mem_ptr(in, &mem_buffer);
+    if (rv <= 0)
+    {
```

Comment:
```suggestion
    if (rv <= 0) {
```

Reply:
OK, but why is leaving an empty file after an empty or mismatched password a greater problem than doing the same for various other reasons? The original issue was specifically about lack of atomicity, leaving an incompletely updated file in place, breaking dependent make rules.

Reply:
Updated.